src/mlpack/tests/random_forest_regressor_test.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file tests/decision_tree_regressor_test.cpp
+ * @author Dinesh Kumar
+ *
+ * Tests for the DecisionTreeRegressor class and related classes.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/random_forest.hpp>
+
+#include "catch.hpp"
+#include "serialization.hpp"
+#include "mock_categorical_data.hpp"
+#include "test_function_tools.hpp"
+
+using namespace mlpack;
+
+/**
+ * Make sure an empty forest cannot predict.
+ */
+TEST_CASE("EmptyPredictTest", "[RandomForestRegressorTest]")
+{
+  RandomForestRegressor<> rf; // No training.
+
+  arma::mat points(10, 100, arma::fill::randu);
+  arma::Row<double> predictions;
+  REQUIRE_THROWS_AS(rf.Predict(points, predictions), std::invalid_argument);
+  REQUIRE_THROWS_AS(rf.Predict(points.col(0)), std::invalid_argument);
+}
+
+/**
+ * Test unweighted numeric learning, making sure that we get better performance
+ * than a single decision tree.
+ */
+TEST_CASE("UnweightedLearnTest", "[RandomForestRegressorTest]")
```

Comment:
Do you think you can make this a `TEMPLATE_TEST_CASE` and test on floating-point data too, like `arma::fmat`?